### PR TITLE
Addresses changes made to security demo config install tool

### PR DESCRIPTION
**Pull request by rishabh6788**
### Description
There were recent changes to security demo configuration setup which now requires a custom admin Password to be setup. This PR updates the documentation and related code change.  

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
-- [Check Logs](#check-logs)
-- [Access EC2 Instances](#access-ec2-instances)
-- [Port Mapping](#port-mapping)
-- [Teardown](#teardown)
-- [Contributing](#contributing)
-- [Getting Help](#getting-help)
-- [Code of Conduct](#code-of-conduct)
-- [Security](#security)
-- [License](#license)
+- [CDK for deploying single-node and multi-node OpenSearch cluster with OpenSearch Dashboards](#cdk-for-deploying-single-node-and-multi-node-opensearch-cluster-with-opensearch-dashboards)
```
Comment (gaiksaya) [inline README.md:3]: Is this duplicate of line 1? Not required right?

File: lib/infra/infra-stack.ts [hunk 2]
```diff
@@ -658,7 +659,8 @@ export class InfraStack extends Stack {
           ignoreErrors: false,
         }));
     } else {
-      cfnInitConfig.push(InitCommand.shellCommand('set -ex;cd opensearch; sudo -u ec2-user nohup ./opensearch-tar-install.sh >> install.log 2>&1 &',
+      // eslint-disable-next-line max-len
+      cfnInitConfig.push(InitCommand.shellCommand(`set -ex;cd opensearch; sudo -u ec2-user nohup env OPENSEARCH_INITIAL_ADMIN_PASSWORD=${props.adminPassword} ./opensearch-tar-install.sh >> install.log 2>&1 &`,
```
Comment (gaiksaya) [inline lib/infra/infra-stack.ts:663]: What happens for version below 2.12?

File: lib/os-cluster-entrypoint.ts [hunk 3]
```diff
@@ -84,7 +84,13 @@ export class OsClusterEntrypoint {
       if (securityDisabled !== 'true' && securityDisabled !== 'false') {
         throw new Error('securityEnabled parameter is required to be set as - true or false');
       }
-      const security = securityDisabled === 'true';
+      const insecure = securityDisabled === 'true';
+
+      const adminPassword: String = insecure ? '' : `${scope.node.tryGetContext('adminPassword')}`;
+      console.log(`Admin password is ${adminPassword}`);
```
Comment (gaiksaya) [inline lib/os-cluster-entrypoint.ts:90]: Debug console statement? Printing password to the console can be risky esp when it comes to deployments.

File: lib/infra/infra-stack.ts (hunk 2 above)
Comment (gaiksaya) [reply lib/infra/infra-stack.ts:663]: Got my answer if the later code

File: lib/os-cluster-entrypoint.ts (hunk 3 above)
Comment (rishabh6788) [reply lib/os-cluster-entrypoint.ts]: Added it for testing, missed to remove it. Removed.

File: README.md (hunk 1 above)
Comment (rishabh6788) [reply README.md]: VS code editor added it for some reason. Removed it.

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**